src/Xamarin.Android.Build.Tasks/Tasks/Crunch.cs

```diff
@@ -9,86 +9,158 @@
 using System.Text.RegularExpressions;
 using Xamarin.Android.Build.Utilities;
 using Xamarin.Android.Tools.Aidl;
+using ThreadingTasks = System.Threading.Tasks;
 
 namespace Xamarin.Android.Tasks
 {
-	public class Crunch : ToolTask
+	public class Crunch : AsyncTask
 	{
+		private const string error_regex_string = @"(?<file>.*)\s*:\s*(?<line>\d*)\s*:\s*error:\s*(?<error>.+)";
```

Comment:
Do we have examples of expected input to match? I'm -- *personally* -- not very keen on `.*`, and this regex would match `::error:string`, which seems rather non-sensical. Perhaps it should be:

    @"(?<file>[^\s]+)\s*:\s*(?<line>\d*)\s*:\s*error:\s*(?<error>.+)

Though that still allows an "empty" line number, which would result in `::` for the matched text. Is that a plausible thing to match?